### PR TITLE
Add Rustup installation instructions to README

### PR DESCRIPTION
**Pull request by Kobzol**
Now that cranelift is distributed with Rustup, I find myself constantly searching for the https://github.com/rust-lang/rust/pull/81746 PR. I think that it would be useful to also describe the Rustup installation/usage instructions here in README.

### CODE REVIEW

**Comment by lqd** [comment]
https://bjorn3.github.io/2023/10/31/progress-report-oct-2023.html describes the way to use the backend via `Cargo.toml` in order to not change the compilation commands (and also via cargo's env vars), which could be interesting here as well.

**Comment by lqd** [comment]
Maybe we should also mention it in https://nnethercote.github.io/perf-book/compile-times.html

**Comment by lqd** [comment]
And also document this in the unstable book.

**Comment by Kobzol** [comment]
Added more usage instructions based on the blog post. Also sent PRs to the perf book (https://github.com/nnethercote/perf-book/pull/72).

I'm not sure if the unstable book is the correct place for this, as it contains just general descriptions of compiler flags. I think that the Rustc book might be more appropriate.

File: Readme.md [hunk 1]
```diff
@@ -5,6 +5,34 @@ This has the potential to improve compilation times in debug mode.
 If your project doesn't use any of the things listed under "Not yet supported", it should work fine.
 If not please open an issue.
 
+## Download using Rustup
+The Cranelift codegen backend is now distributed in nightly builds on Linux. If you want to
```
Comment (bjorn3) [inline Readme.md:9]: and x86_64 macOS

**Comment by bjorn3** [comment]
The actual text of the new section looks good to me. It feels a bit weird to have using precompiled versions from rustup first, then manual building and finally using precompiled versions from CI though.

File: Readme.md (hunk 1 above)
Comment (Kobzol) [reply Readme.md:9]: Thanks, fixed.

**Comment by Kobzol** [comment]
I agree. I think that it makes sense to have the easiest way first though. What order would you prefer?

**Comment by bjorn3** [comment]
Maybe swap building it yourself and using the CI artifacts? You would need to reword using the CI artifacts for that as it is currently written as alternative for building yourself.

**Comment by Kobzol** [comment]
I tried to reorganize it a little bit.

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**